parsec/api/protocol/block.py

```diff
@@ -1,100 +1,48 @@
-    BaseTypedReqSchema,
-    BaseTypedRepSchema,
-    CmdSerializer,
-    BaseReq,
-    BaseRep,
-    cmd_rep_error_type_factory,
-    cmd_rep_factory,
-)
-from parsec.api.protocol.realm import RealmIDField
 from parsec._parsec import BlockID
 
 __all__ = ("BlockID", "BlockIDField", "block_create_serializer", "block_read_serializer")
 
 
 BlockIDField: Type[fields.Field] = fields.uuid_based_field_factory(BlockID)
 
+from parsec._parsec import (
+    BlockReadReq,
+    BlockReadRep,
+    BlockCreateReq,
+    BlockCreateRep,
+)
 
-class BlockCreateReqSchema(BaseReqSchema):
-    block_id = BlockIDField(required=True)
-    realm_id = RealmIDField(required=True)
-    block = fields.Bytes(required=True)
-
-
-class BlockCreateRepSchema(BaseRepSchema):
-    pass
-
-
-block_create_serializer = CmdSerializer(BlockCreateReqSchema, BlockCreateRepSchema)
-
-
-@attr.s(slots=True, frozen=True, auto_attribs=True, kw_only=True, eq=False)
-class BlockReadReq(BaseReq):
-    class SCHEMA_CLS(BaseTypedReqSchema):
-        cmd = fields.CheckedConstant("block_read", required=True)
-        block_id = BlockIDField(required=True)
-
-    block_id: BlockID
-
-
-@attr.s(slots=True, frozen=True, auto_attribs=True, kw_only=True, eq=False)
-class BlockReadRepOk(BaseRep):
-    class SCHEMA_CLS(BaseTypedRepSchema):
-        status = fields.CheckedConstant("ok", required=True)
-        block = fields.Bytes(required=True)
-
-    block: bytes
 
+class BlockCreateSerializer:
+    @staticmethod
+    def req_dumps(req) -> bytes:  # type: ignore[no-untyped-def]
+        return BlockCreateReq(req["block_id"], req["realm_id"], req["block"]).dump()
 
-BlockReadRepNotFound = cmd_rep_error_type_factory("BlockReadRepNotFound", "not_found")
-BlockReadRepTimeout = cmd_rep_error_type_factory("BlockReadRepTimeout", "timeout")
-BlockReadRepNotAllowed = cmd_rep_error_type_factory("BlockReadRepNotAllowed", "not_allowed")
-BlockReadRepInMaintenance = cmd_rep_error_type_factory(
-    "BlockReadRepInMaintenance", "in_maintenance"
-)
+    @staticmethod
+    def rep_loads(raw: bytes) -> BlockCreateRep:
+        try:
+            return BlockCreateRep.load(raw)
+        except ValueError as exc:
+            raise exc
 
-BlockReadRep = cmd_rep_factory(
-    "BlockReadRep",
-    BlockReadRepOk,
-    BlockReadRepNotFound,
-    BlockReadRepTimeout,
-    BlockReadRepNotAllowed,
-    BlockReadRepInMaintenance,
-)
 
-BlockReadRepType = Union[
-    BlockReadRepOk,
-    BlockReadRepNotFound,  # type: ignore[valid-type]
-    BlockReadRepTimeout,  # type: ignore[valid-type]
-    BlockReadRepNotAllowed,  # type: ignore[valid-type]
-    BlockReadRepInMaintenance,  # type: ignore[valid-type]
-]
+class BlockReadSerializer:
+    @staticmethod
+    def req_dumps(req) -> bytes:  # type: ignore[no-untyped-def]
+        return BlockReadReq(req["block_id"]).dump()
 
-block_read_serializer = CmdSerializer.from_typed(BlockReadReq, BlockReadRep.TYPES)
+    @staticmethod
+    def rep_loads(raw: bytes) -> BlockReadRep:
```

Comment:
We only implement `req_dumps` and `rep_loads` because the server side no longer uses this class ? (and soon we will have the backend_connection client part in Rust and this serializer would be entirely removed \o/)

Reply:
I don't know :man_shrugging: but yes, many things can be removed IMO